source/snake-case.d.ts

```diff
@@ -0,0 +1,30 @@
+import {DelimiterCase} from './delimiter-case';
+
+/**
+Converts a string literal from any typical non-snake-case casing to a snake-case
```

Comment:
Is it clear to the user what "typical non-snake-case casing" means?

Reply:
We could simplify  it to:

> Converts a string literal to snake case

Alternative would be to list them all:

> Converts a kebab cased, camel cased and/or pascal cased string literal to snake case

Reply:
Let's go with the former